csrc/dynamic_transform.cpp

```diff
@@ -940,13 +941,30 @@ void DynamicTransformConcretizer::concretizeResize() {
         id->definition() && id->definition()->isA<Resize>(),
         "Resized IterDomain must have a Resize definition");
     auto def = id->definition()->as<Resize>();
+    auto def_in = maybeMutated(def->in())->as<IterDomain>();
     auto new_id = IterDomain::resize(
-        def->in(),
-        def->leftExpand(),
-        def->rightExpand(),
+        def_in,
+        maybeMutated(def->leftExpand()),
+        maybeMutated(def->rightExpand()),
         id->isRFactorProduct(),
         iter_type);
 
+    if (maybeMutated(def_in->extent()) != def_in->extent()) {
```

Comment:
If the extent is mutated, doesn't that mean `def_in` itself is also mutated? If so, the above call to `IterDomain::resize` uses the mutated `def_in`, so is this block of the code still required?